### ISSUE TITLE
[Feature] Support for Ivy & Angular 16

### ISSUE THREAD

**Issue by VeitWeber** [stephanrauh/ngx-extended-pdf-viewer#1704]
With Angular 16 [Angular Compatibility Compiler (ngcc) will be removed](https://github.com/angular/angular/releases/tag/16.0.0-next.0) and as a result ngx-extend-pdf-viewer will no longer work. Are there any plans to migrate the lib to Ivy? I'm glad to help if needed.

**Comment by stephanrauh** [comment]
You're surprising me. My library is still compatible to old renderer? I didn't expect that.

I've migrated to Ivy a long time ago. If you run into problems, just tell me, but I'm pretty sure you can update without risk.

I'm keeping this ticket open for now because I want to check if I'm still compiling for both Ivy and pre-Ivy. If so, I'll have to remove that configuration soon.

**Comment by VeitWeber** [comment]
Hi Stephan, didn't dive into the code just saw this on the latest Angular and ngx-extended-pdf-viewer. 

⠙ Generating browser application bundles (phase: setup)...Processing legacy "View Engine" libraries:

`ngx-extended-pdf-viewer [es2015/esm2015] (stephanrauh/ngx-extended-pdf-viewer.git)
Encourage the library authors to publish an Ivy distribution.`

Maybe it's a config issue on my side. Will check this tomorrow.

**Comment by stephanrauh** [comment]
... and it was very kind of you to encourage me. Highly appreciated! Few people do so, and that's a pity.

I've just compiled the showcase. I don't see the warning:
```
ng b --configuration production
✔ Browser application bundle generation complete.
✔ Copying assets complete.
✔ Index html generation complete.

Initial Chunk Files           | Names                                          |  Raw Size | Estimated Transfer Size
main.a75fccb671b5a4a3.js      | main                                           |   1.39 MB |               295.48 kB
styles.12c2f895ede1ca74.css   | styles                                         |  84.65 kB |                 9.47 kB
scripts.54984343f5a3c709.js   | scripts                                        |  70.86 kB |                20.40 kB
polyfills.36261152741200d8.js | polyfills                                      |  33.12 kB |                10.66 kB
runtime.fc32f15aa627ecaf.js   | runtime                                        |   2.72 kB |                 1.28 kB

                              | Initial Total                                  |   1.57 MB |               337.29 kB

Lazy Chunk Files              | Names                                          |  Raw Size | Estimated Transfer Size
803.046648dfc5ac86d8.js       | extended-pdf-viewer-extended-pdf-viewer-module | 527.05 kB |                95.93 kB

Build at: 2023-03-07T21:48:49.757Z - Hash: 61a09f903dea04f0 - Time: 21562ms

Warning: /Users/stephanrauh/git/pdf/extended-pdf-viewer-showcase/node_modules/angular2-multiselect-dropdown/__ivy_ngcc__/fesm2015/angular2-multiselect-dropdown.js depends on '@tweenjs/tween.js'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies
```

I wonder what triggers the warning. If you create a new sandbox application (`ng new demo`, `cd demo`, `ng add ngx-extended-pdf-viewer`, `ng s -o`), does it show the warning, too?

**Comment by VeitWeber** [comment]
Sandbox worked fine and now I can't even reproduce it in the App that generates the warning yesterday.  ¯\_(ツ)_/¯ Sorry for wasting your time.

**Comment by stephanrauh** [comment]
Don't worry, it was a pleasure! Actually, I wonder what I need to do to prepare for Angular 16. Currently, a lot of exciting stuff happens in the Angular world: functional guards, standalone components, and probably a couple of things I've missed. Does any of that have any impact on my library?

**Comment by VeitWeber** [comment]
And Signals! I also love the direction Angular is heading at the moment. By the way, I stumbled upon your name today and wondered where I knew it from. We already had some interaction in 2016, back then in the context of Bootsfaces. You gave a presentation at Javaland and I wanted to present Bootsfaces  in our user group. It's funny how we meet again.

**Comment by stephanrauh** [comment]
Indeed, that's funny. Your name sounded familiar, but I was slightly puzzled. Now I know why I was puzzled. :)

BTW, this year I also submitted a talk to Javaland, but I only made it to the replacement list. If I'm very lucky, they may still call me and ask me to talk about AWS CDK. In any case, I'm going to visit Javaland this year in person. If you're there, too, give me a call, and we can have a coffee together.

Another opportunity to meet me virtually is https://entwickler.de/graalvm-day/.  Me first public talk in English. I have to admit I a bit nervous! :)
